server/src/main/kotlin/suwayomi/tachidesk/manga/impl/update/Updater.kt

```diff
@@ -14,31 +14,50 @@ import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Semaphore
+import kotlinx.coroutines.sync.withPermit
 import mu.KotlinLogging
 import suwayomi.tachidesk.manga.impl.Chapter
 import suwayomi.tachidesk.manga.model.dataclass.MangaDataClass
 import java.util.concurrent.ConcurrentHashMap
 
 class Updater : IUpdater {
+    companion object {
+        const val MAX_PARALLEL_UPDATE_REQUESTS = 10
+    }
+
     private val logger = KotlinLogging.logger {}
     private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Default)
 
     private val _status = MutableStateFlow(UpdateStatus())
     override val status = _status.asStateFlow()
 
     private val tracker = ConcurrentHashMap<Int, UpdateJob>()
-    private var updateChannel = createUpdateChannel()
+    private val updateChannels = mutableMapOf<String, Channel<UpdateJob>>()
```

Comment:
Should also probably use a ConcurrentMap so that we don't run into issues with multiple channel creation
```suggestion
    private val updateChannels = ConcurrentHashMap<String, Channel<UpdateJob>>()
```

Reply:
done